File: JUnitTestingLastAct/src/Strategy/StrategyTestRunner.java
package Strategy;

import Singleton.SingletonTestRunner;
import org.testng.annotations.Test;

import java.util.logging.Logger;

import static org.testng.AssertJUnit.assertEquals;

public class StrategyTestRunner {

    private final Logger logger = Logger.getLogger(StrategyTestRunner.class.getName());
    @Test
    public final void checkSum(){
        Context context = new Context(new OperationAdd());
        int sum = StrategyTestCases.correctSum(1,2);
        assertEquals(sum,context.executeStrategy(1,2));
        this.logger.info("The sum produced is equal");

    }

    @Test
    public final void checkDifference(){
        Context context = new Context(new OperationSubtract());
        int diff = StrategyTestCases.correctDifference(5,3);
        assertEquals(diff,context.executeStrategy(5,3));
        this.logger.info("The difference produced is equal");
    }

    @Test
    public final void checkProduct(){
        Context context = new Context(new OperationMultiply());
        int product = StrategyTestCases.correctProduct(5,3);
        assertEquals(product,context.executeStrategy(5,3));
        this.logger.info("The product produced is equal");
    }

}

File: KotlinTesting/src/test/behavioral/StrategyTestRunner.kt
package Strategy

import Strategy.StrategyTestCases.correctDifference
import Strategy.StrategyTestCases.correctProduct
import Strategy.StrategyTestCases.correctSum
import junit.framework.Assert.assertEquals
import org.junit.Test

import java.util.logging.Logger

class StrategyTestRunner {
    private val logger = Logger.getLogger(StrategyTestRunner::class.java.name)
    @Test
    fun checkSum() {
        val context = Context(OperationAdd())
        val sum = correctSum(1, 2)
        assertEquals(sum, context.executeStrategy(1, 2))
        logger.info("The sum produced is equal")
    }

    @Test
    fun checkDifference() {
        val context = Context(OperationSubtract())
        val diff = correctDifference(5, 3)
        assertEquals(diff, context.executeStrategy(5, 3))
        logger.info("The difference produced is equal")
    }

    @Test
    fun checkProduct() {
        val context = Context(OperationMultiply())
        val product = correctProduct(5, 3)
        assertEquals(product, context.executeStrategy(5, 3))
        logger.info("The product produced is equal")
    }
}